From: Yury Norov <yury.norov@gmail.com>
Subject: lib/group_cpus: relax atomicity requirement in grp_spread_init_one()
Date: Thu, 28 Dec 2023 12:09:30 -0800

Because nmsk and irqmsk are stable, extra atomicity is not required.

__cpumask_clear_cpu() and __cpumask_set_cpu() are added in commit
6c8557bdb28d ("smp, cpumask: Use non-atomic cpumask_{set,clear}_cpu()")
for fast code path( smp_call_function_many()).

We have ~670 users of cpumask_clear_cpu & cpumask_set_cpu, lots of them
fall into same category with group_cpus.c(doesn't care atomicity, not in
fast code path), and needn't change to __cpumask_clear_cpu() and
__cpumask_set_cpu().  Otherwise, this way may encourage to update others
into the __cpumask_* version.

Link: https://lkml.kernel.org/r/20231228200936.2475595-4-yury.norov@gmail.com
Signed-off-by: Yury Norov <yury.norov@gmail.com>
NAKed-by: Ming Lei <ming.lei@redhat.com>
Cc: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Cc: Rasmus Villemoes <linux@rasmusvillemoes.dk>
Cc: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/group_cpus.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/lib/group_cpus.c~lib-group_cpus-relax-atomicity-requirement-in-grp_spread_init_one
+++ a/lib/group_cpus.c
@@ -24,8 +24,8 @@ static void grp_spread_init_one(struct c
 		if (cpu >= nr_cpu_ids)
 			return;
 
-		cpumask_clear_cpu(cpu, nmsk);
-		cpumask_set_cpu(cpu, irqmsk);
+		__cpumask_clear_cpu(cpu, nmsk);
+		__cpumask_set_cpu(cpu, irqmsk);
 		cpus_per_grp--;
 
 		/* If the cpu has siblings, use them first */
@@ -36,8 +36,8 @@ static void grp_spread_init_one(struct c
 			if (cpus_per_grp-- == 0)
 				return;
 
-			cpumask_clear_cpu(sibl, nmsk);
-			cpumask_set_cpu(sibl, irqmsk);
+			__cpumask_clear_cpu(sibl, nmsk);
+			__cpumask_set_cpu(sibl, irqmsk);
 		}
 	}
 }
_
